python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -830,6 +836,55 @@ def addDimensionEntry(self, dimension, dataId=None, entry=None, **kwds):
             self.setDimensionRegion(dataId)
         return dataId
 
+    @disableWhenLimited
+    @transactional
+    def addDimensionEntryList(self, dimension, dataIdList, entry=None, **kwds):
+        # Docstring inherited from Registry.addDimensionEntryList.
+        dataIdList = [DataId(dataId, dimension=dimension, universe=self.dimensions, **kwds) for dataId in
+                      dataIdList]
+        # The given dimension should be the only leaf dimension of the graph,
+        # and this should ensure it's a true `Dimension`, not a `str` name.
+        # all of dataIds should share a common dimension, so use the first
+        dimension, = dataIdList[0].dimensions().leaves
+        if entry is not None:
+            for dataId in dataIdList:
+                dataId.entries[dimension].update(entry)
+        table = self._schema.tables[dimension.name]
+        if table is None:
+            raise TypeError(f"Dimension '{dimension.name}' has no table.")
+        holder = dataIdList[0].dimensions().getRegionHolder()
+        if holder is not None:
+            # Update the join table between this Dimension and SkyPix, if it
+            # isn't itself a view.
+            skypixJoin = dataIdList[0].dimensions().union(["SkyPix"]).joins().findIf(
+                lambda join: join != holder and join.name not in self._schema.views
+            )
+        else:
+            skypixJoin = None
+        chunks = [dataIdList[i:i+self.defaultChunkSize] for i in range(0, len(dataIdList),
+                  self.defaultChunkSize)]
+        skypixParams = []
+        if skypixJoin is not None:
+            for dataId in dataIdList:
+                if dataId.region is not None:
+                    for begin, end in self.pixelization.envelope(dataId.region).ranges():
+                        for skypix in range(begin, end):
+                            skypixParams.append(dict(dataId, skypix=skypix))
+        skypixParamChunks = [skypixParams[i:i+self.defaultChunkSize] for i in range(0, len(skypixParams),
+                             self.defaultChunkSize)]
+        try:
+            for i, chunk in enumerate(chunks):
+                self._connection.execute(table.insert().values([dataId.fields(dimension, region=True) for
+                                                                dataId in chunk]))
+
+        except IntegrityError:
+            # TODO check for conflict, not just existence.
+            raise ConflictingDefinitionError(f"Existing definition for {dimension.name} entry in {chunk}.")
+        # Insert all of the cunked up skypix entries into the skypix join table
+        for i, chunk in enumerate(skypixParamChunks):
+            self._connection.execute(self._schema.tables[skypixJoin.name].insert(), skypixParams)
```

Comment:
This doesn't raise here or in `setDimensionRegion`, but I'm not clear why.  Is attempting to set the region in a repo that doesn't have a skymap/skypix over that region not an error?